File: Final Project/Grocery Application/Grocery-API/Grocery/Model/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Grocery.Model
{
    public class Order
    {
        [Key]
        public int OrderId { get; set; }

        
        public int CustomerId { get; set; }  // Foreign key to Customer

        
        public DateTime OrderDate { get; set; }

        
        [StringLength(20)]
        public string? Status { get; set; }  // E.g., Pending, Completed, Delivered

        public int? DeliveryStaffId { get; set; }  // Nullable foreign key to DeliveryStaff

        public Customer Customer { get; set; }  // Navigation property to Customer
        public DeliveryStaff? DeliveryStaff { get; set; }  // Navigation property to DeliveryStaff
        public ICollection<OrderItem>? OrderItems { get; set; }  // Navigation property to OrderItems

        // Calculated property to get the total price
        [NotMapped]
        public decimal? TotalPrice;
    }

    public class OrderUpdateModel
    {
        public int OrderId { get; set; }
        public DateTime OrderDate { get; set; }
        public int CustomerId { get; set; }

        [StringLength(20)]
        public string? Status { get; set; }  // E.g., Pending, Completed, Delivered

        public int? DeliveryStaffId { get; set; }
    }
}

File: Final Project/Grocery Application/Grocery-Angular/src/Models/Order.ts
import { User } from "./User";

export interface Order {
    orderId: number;
    customerId: number;
    orderDate: Date;
    status?: string;
    deliveryStaffId?: number;
    totalPrice?: number; // Calculated property
    customer?: Customer; // Navigation property
    deliveryStaff?: DeliveryStaff; // Navigation property
    orderItems?: OrderItem[]; // Navigation property
  }
  
  export interface Customer {
    customerId: number;  
    name?: string;
    userId:number;
    address?: string;
    phoneNumber?:string;
    user:User
  }
  
  export interface DeliveryStaff {
    deliveryStaffId: number;
    name: string;
    phoneNumber:number;
  }
  
  export interface OrderItem {
    orderItemId: number;
    productId: number;
    quantity: number;
    price: number;
  }
  